tests/test_core.py

```diff
@@ -72,31 +72,30 @@
 
 
 class TestParserSelection:
-    def test_selects_parsers(self):
-        for fp in BL1_files:
-            parser = bletl.get_parser(fp)
-            assert isinstance(parser, core.BLDParser)
-            assert isinstance(parser, parsing.bl1.BioLector1Parser)
-
+    @pytest.mark.parametrize("fp", BL1_files)
+    def test_selects_parsers(self, fp):
+        parser = bletl.get_parser(fp)
+        assert isinstance(parser, core.BLDParser)
+        assert isinstance(parser, parsing.bl1.BioLector1Parser)
         return
 
     def test_fail_on_unsupported(self):
         with pytest.raises(ValueError):
             bletl.get_parser(not_a_bl_file)
         return
 
-    def test_selects_parsers_pro(self):
-        for fp in BLPro_files:
-            parser = bletl.get_parser(fp)
-            assert isinstance(parser, core.BLDParser)
-            assert isinstance(parser, parsing.blpro.BioLectorProParser)
+    @pytest.mark.parametrize("fp", BLPro_files)
+    def test_selects_parsers_pro(self, fp):
+        parser = bletl.get_parser(fp)
+        assert isinstance(parser, core.BLDParser)
+        assert isinstance(parser, parsing.blpro.BioLectorProParser)
         return
 
-    def test_selects_parsers_ii(self):
-        for fp in BL2_files:
-            parser = bletl.get_parser(fp)
-            assert isinstance(parser, core.BLDParser)
```

Comment:
When you change this where is now the definition for fp? Before it was extracted from the BioLector_files. fp is passed to the function in your new commits but I do not get where it is defined. Or do you now need a pre definition for every use of the functions?

Reply:
The `parametrize` decorator passes them as the `fp` keyword agument:
```python
@pytest.mark.parametrize("fp", BL2_files)
def test_selects_parsers_ii(self, fp):
```